src/metabase/driver/bigquery.clj

```diff
@@ -351,7 +351,7 @@
         results (if mbql?
                   (post-process-mbql dataset-id table-name results)
                   (update results :columns (partial map keyword)))]
-    (assoc results :annotate? true)))
+    (assoc results :annotate? mbql?)))
```

Comment:
I would add a comment here but that logic is explained in a few other places in the code so I don't think it is necessary to repeat the explanation here